File: clojure-examples/src/code_examples/fizz_buzz.clj
(ns code-examples.fizz-buzz)

(defn divisible-by? [divisor number]
  (= 0 (mod number divisor)))

(defn fizz-buzzify [number]
  (cond
    (divisible-by? 15 number) "fizzbuzz"
    (divisible-by? 3 number) "fizz"
    (divisible-by? 5 number) "buzz"
    :else number))

(def fizz-buzz (map fizz-buzzify (range)))

(mapv println (take 100 fizz-buzz))

File: ruby-examples/fizz_buzz.rb
number = 1

while number < 20
  if number % 3 == 0 && number % 5 == 0
    puts "fizzbuzz"
  elsif number % 3 == 0
    puts "fizz"
  elsif number % 5 == 0
    puts "buzz"
  else
    puts number
  end

  number = number + 1
end

#########################################

for number in 1..20
  if number % 3 == 0 && number % 5 == 0
    puts "fizzbuzz"
  elsif number % 3 == 0
    puts "fizz"
  elsif number % 5 == 0
    puts "buzz"
  else
    puts number
  end
end

#########################################

numbers = (0..20).map do |number|
  if number % 3 == 0 && number % 5 == 0
    "fizzbuzz"
  elsif number % 3 == 0
    "fizz"
  elsif number % 5 == 0
    "buzz"
  else
    number
  end
end

puts numbers
